### ISSUE TITLE
Optional syntax throws "no-undef" in 8.2.2+

### ISSUE THREAD

**Issue by gabaum10** [babel/babel-eslint#633]
Take the following code:

```
   const test = {
        a: 'test string'
    };

    console.log(test.a?.b);
```

In `babel-eslint@8.2.2`+ b is throwing a `no-undef` error.  Downgrading to 8.2.1 does not.   Previously reported in #511

**Comment by richardgirges** [comment]
Still seeing this behavior as well on every version of babel-eslint after 8.2.1 (including the master branch). Not sure why #511 was closed. Downgrading doesn't seem like an acceptable solution.

**Comment by lehni** [comment]
This got addressed in #595, but I am seeing new problems with 8.2.4 and 8.2.5 now.

**Comment by lehni** [comment]
@existentialism you can probably close this issue here.
